### PR TITLE
Sprinkler - Completed timer functionality & optimised logging

### PR DESCRIPTION
**Pull request by AndreasPrang**
The HomeKit timer functionality was not really implemented. Now it works.

### CODE REVIEW

**Comment by AndreasPrang** [comment]
Fixed status updates.

**Comment by KhaosT** [comment]
Thanks for contributing. However, the goal of those accessory examples is to show how to use the services, instead of making them simulate everything... Adding timer support introduces additional complexity that makes the control flow hard to follow as a basic example.

If you believe otherwise, feel free to reopen the PR.
